ml_engine/online_prediction/predict.py

```diff
@@ -0,0 +1,173 @@
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+"""Examples of using the Cloud ML Engine's online prediction service."""
+from __future__ import print_function
+# [START import_libraries]
+import googleapiclient.discovery
+# [END import_libraries]
+
+
+# [START authenticating]
+def get_ml_engine_service():
+    """Create the ML Engine service object.
+    To authenticate set the environment variable
+    GOOGLE_APPLICATION_CREDENTIALS=<path_to_service_account_file>
+    """
+    return googleapiclient.discovery.build('ml', 'v1beta1')
+# [END authenticating]
+
+
+# [START predict_json]
+def predict_json(project, model, instances, version=None):
+    """Send data instances to a deployed model for prediction
+    Args:
+        project: str, project where the Cloud ML Engine Model is deployed.
+        model: str, model name.
+        instances: [dict], dictionaries from string keys defined by the model
+        to data.
+        version: [optional] str, version of the model to target.
+    Returns:
+        A dictionary of prediction results defined by the model.
+    """
+    service = get_ml_engine_service()
+    name = 'projects/{}/models/{}'.format(project, model)
+    if version is not None:
+        name += '/versions/{}'.format(version)
+
+    response = service.projects().predict(
+        name=name,
+        body={'instances': instances}
+    ).execute()
+
+    if 'error' in response:
+        raise RuntimeError(response['error'])
+
+    return response['predictions']
+# [END predict_json]
+
+
+# [START predict_tf_records]
+def predict_tf_records(project,
+                       model,
+                       example_bytes_list,
+                       key='tfrecord',
+                       version=None):
+    """Send data instances to a deployed model for prediction
+    Args:
+        project: str, project where the Cloud ML Engine Model is deployed
+        model: str, model name.
+        example_bytes_list: [str], Serialized tf.train.Example protos.
+        version: str, version of the model to target.
+    Returns:
+        A dictionary of prediction results defined by the model.
+    """
+    import base64
+    service = get_ml_engine_service()
+    name = 'projects/{}/models/{}'.format(project, model)
+    if version is not None:
+        name += '/versions/{}'.format(version)
+
+    response = service.projects().predict(
+        name=name,
+        body={'instances': [
+            {key: {'b64': base64.b64encode(example_bytes)}}
+            for example_bytes in example_bytes_list
+        ]}
+    ).execute()
+    if 'error' in response:
+        raise RuntimeError(response['error'])
+
+    return response['predictions']
+# [END predict_tf_records]
+
+
+# [START census_to_example_bytes]
+def census_to_example_bytes(json_instance):
```

Comment:
I was expecting the file path to be in the census example (in cloudml-samples). Is this file part of the census sample or a more generic 'calling online prediction' sample? If the latter, we need better warnings that this will not work with every model, and we need to describe what the model is expecting.

If this is not part of the census sample, a s/census/json/g is needed.

Sorry if this is a bad question, I not familiar with python-docs-samples